### PR TITLE
AI-458: Align POOL Team Lifecycle: No Trial Expiry, No Retention Deletion

### PR DESCRIPTION
**Pull request by dspachos**
<details><summary><h3>Greptile Summary</h3></summary>

This PR aligns the POOL team lifecycle so that all POOL teams are fully exempt from inactivity-based retention deletion and trial-expiry key expiration, regardless of whether they have a purchase or have `require_purchase_for_requests` set.

- `_check_team_retention_policy` now short-circuits on `budget_type == POOL`, preventing warning emails and soft-deletion for any POOL team; a new test covers the case where `retention_warning_sent_at` is already set.
- `_send_expiry_notification` drops its `is_pool_team` parameter; the guard moves to the call site in `monitor_teams` (`if not is_pool_team`), and key expiry now uses the same broad `is_pool_team` flag instead of the narrower `has_active_pool_purchase` DB query.
- Stripe reconciliation is selectively skipped only for purchase-gated POOL teams (`requires_pool_purchase_gate`), leaving non-purchase-gated POOL teams still reconciled — consistent with the comment that they may hold Stripe product associations.
</details>

<details open><summary><h3>Confidence Score: 5/5</h3></summary>

Safe to merge — the core logic change is correct and the retention/expiry exemption for all POOL teams is consistently applied across the three guard points.

The three guard points (retention, expiry notification, key expiry) are now all consistently keyed on team.budget_type == BudgetType.POOL, and the new retention test covers the edge case where retention_warning_sent_at is already populated. The Stripe reconciliation guard stays narrower (requires_pool_purchase_gate) by design. The only gap is that neither worker test exercises the POOL + require_purchase_for_requests=False combination for key-expiry exemption, but the production code path is straightforward and the flag used is the same one tested in the retention suite.

The test coverage in tests/test_worker.py does not exercise POOL teams where require_purchase_for_requests is False; a future reader changing the guard expression could regress that path silently.
</details>

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| app/core/worker.py | Core logic change: POOL teams are now fully exempt from retention deletion and trial expiry; _send_expiry_notification loses its is_pool_team param, guard is lifted to the call site; key-expiry check switches from has_active_pool_purchase DB query to the broader is_pool_team flag; Stripe reconciliation is selectively skipped only for purchase-gated POOL teams. |
| tests/test_team_retention.py | New test verifies POOL teams are fully skipped by the retention policy even when retention_warning_sent_at is already set, covering the early-return guard added to _check_team_retention_policy. |
| tests/test_worker.py | Tests updated to reflect new all-POOL-teams-exempt-from-trial-expiry policy; both scenarios add require_purchase_for_requests=True which implicitly skips Stripe reconciliation, leaving the POOL+require_purchase_for_requests=False branch untested for key-expiry exemption. |
| scripts/initialise_resources.py | Cosmetic reformatting only (line-length wrapping), no functional changes. |

</details>

</details>

<details><summary><h3>Flowchart</h3></summary>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[monitor_teams: iterate team] --> B{requires_pool_purchase_gate?}
    B -- Yes --> C[Skip Stripe reconciliation]
    B -- No --> D[reconcile_team_product_associations]
    C --> E[Check has_products]
    D --> E
    E --> F[_check_team_retention_policy]
    F --> G{budget_type == POOL?}
    G -- Yes --> H[Return early — no retention action]
    G -- No --> I[Evaluate inactivity / warning / soft-delete]
    H --> J[Compute team_freshness metrics]
    I --> J
    J --> K{budget_type == POOL?}
    K -- Yes --> L[Skip _send_expiry_notification]
    K -- No --> M[_send_expiry_notification]
    L --> N{budget_type == POOL?}
    M --> N
    N -- Yes --> O[Skip key expiry]
    N -- No --> P{not has_products AND days_remaining <= 0?}
    P -- Yes --> Q[expire_keys = True]
    P -- No --> R[expire_keys = False]
    O --> S[reconcile_team_keys]
    Q --> S
    R --> S
```
</details>

<sub>Reviews (2): Last reviewed commit: ["refactor(worker): simplify pool team not..."](https://github.com/amazeeio/amazee.ai/commit/7d7b69a4f9d2edd233b6c2f4d5f65d40a14b9b89) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=31920566)</sub>

### CODE REVIEW

File: app/core/worker.py [hunk 1]
```diff
@@ -1072,35 +1074,31 @@ async def monitor_teams(db: Session):
...
-                    days_remaining,
-                    ses_service,
-                    is_pool_team=team.budget_type == BudgetType.POOL,
-                )
+                # POOL teams are managed by pool-specific lifecycle jobs;
+                # skip trial freshness and expiry notifications.
+                if not is_pool_team:
+                    team_freshness = _monitor_team_freshness(team, db)
+                    days_remaining = TRIAL_OVER_DAYS - team_freshness
+                    _send_expiry_notification(
```
[the remark this answers is not in the document]
Comment (dspachos) [reply app/core/worker.py]: @copilot apply changes based on [this feedback](https://github.com/amazeeio/amazee.ai/pull/490#discussion_r3231854621)

File: app/core/worker.py [hunk 2]
```diff
@@ -1072,35 +1078,30 @@ async def monitor_teams(db: Session):
...
-                    is not None
-                )
-
                 # Expire if team trial has expired (if team has a product, expiry will be handled by Stripe)
-                # Pool teams with purchases are exempt — they are not trial users.
+                # Purchase-gated POOL teams are always exempt from trial expiration.
                 if (
                     not has_products
-                    and not has_active_pool_purchase
+                    and not is_purchase_gated_pool_team
```
[the remark this answers is not in the document]
Comment (dspachos) [reply app/core/worker.py]: Fixed by making trial notification/expiry guards consistent with policy: all POOL teams are excluded from trial lifecycle logic (not only purchase-gated POOL).

File: app/core/worker.py [hunk 3]
```diff
@@ -1072,35 +1078,30 @@ async def monitor_teams(db: Session):
...
+                if not is_purchase_gated_pool_team:
+                    _send_expiry_notification(
+                        db,
+                        team,
+                        has_products,
+                        should_send_notifications,
+                        days_remaining,
+                        ses_service,
+                        is_pool_team=False,
+                    )
```
[the remark this answers is not in the document]
Comment (dspachos) [reply app/core/worker.py]: Fixed by removing the redundant is_pool_team parameter from _send_expiry_notification; POOL exclusion is now handled at the call site, so the function signature is no longer misleading.

**Review by pmelab — APPROVED** [review]
